src/order/upper_lower.lean

```diff
@@ -357,18 +404,24 @@ end has_le
 
 namespace upper_set
 section preorder
-variables [preorder α] {a b : α}
+variables [preorder α] {a b : α} {s : set α}
 
-/-- The smallest upper set containing a given element. -/
+/-- Principal upper set; the smallest upper set containing a given element.
+  `set.Ici` as an upper set. -/
 def Ici (a : α) : upper_set α := ⟨Ici a, is_upper_set_Ici a⟩
 
-/-- The smallest upper set containing a given element. -/
+/-- Strict principal upper set; `set.Ioi` as an upper set. -/
 def Ioi (a : α) : upper_set α := ⟨Ioi a, is_upper_set_Ioi a⟩
 
+/-- The smallest upper set containing a given set. `set.upper_set_of` as an upper set -/
+def upper_set_of (s : set α) : upper_set α := ⟨set.upper_set_of s, is_upper_set_upper_set_of s⟩
```

Comment:
My instinct is that this should be the only version of this to exist, and `set.upper_set_of` should be designed to be written as the coercion of this; then `coe_upper_set_of` can say it's equal to `{x | ∃ a ∈ s, a ≤ x}`. I think the opposite is chosen for Ici because they're already used a lot elsewhere, but `upper_set_of` is being made in this PR.
@YaelDillies may also have an opinion on this though